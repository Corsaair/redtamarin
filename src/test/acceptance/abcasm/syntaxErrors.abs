/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
/*
 *  Test various syntax errors.
 */
function script0$init():*
{
	//  Unknown opcode.
	zuul 12 ls "zeeb"

	//  callproperty insn without enough operands
	callproperty  	parseInt 

	//  This is going to show up in a diagnostic
	//  for "callproperty" because the input is free-format
	pushnull

	//  callproperty insn with wrong type operands
	callproperty  	parseInt (44.2)

	//  Too many integer prefixes
	10 11       setlocal 1

	//  pushscope succeeds, but extra_operand is parsed as an insn
	pushscope extra_operand

	// invalid string arg to debugline
	debugline "zerg"

  //  Last insn in the function is invalid.
  foodle
}

function poot()
{
	//  empty function body -- not allowed
}

//  This, on the other hand, is perfectly fine!
