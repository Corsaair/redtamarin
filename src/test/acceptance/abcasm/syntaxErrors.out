abcasm/syntaxErrors.abs

Unable to assemble abcasm/syntaxErrors.abs due to syntax errors:
line 43:1 Unimplemented opcode "zuul"
line 50:1 no viable alternative at input 'pushnull'
line 53:26 no viable alternative at input '44.2'
line 56:4 no viable alternative at input '11'
line 59:11 Unimplemented opcode "extra_operand"
line 62:11 no viable alternative at input '"zerg"'
line 65:2 Unimplemented opcode "foodle"
line 71:0 required (...)+ loop did not match anything at input '}'
